src/lib/hooks/use-best-quote/utils/flashmint.ts

```diff
@@ -126,73 +103,37 @@ async function getEnhancedFlashMintQuote(
       )
       transaction.gas = gas.limit
 
-      const inputTokenAmountUsd =
-        Number.parseFloat(formatWei(inputAmount, inputToken.decimals)) *
-        inputTokenPrice
-      const outputTokenAmountUsd =
-        Number.parseFloat(formatWei(outputAmount, outputToken.decimals)) *
-        outputTokenPrice
-      const quoteAmountUsd =
-        Number.parseFloat(
-          formatWei(
-            quoteAmount,
-            isMinting ? inputToken.decimals : outputToken.decimals,
-          ),
-        ) * (isMinting ? inputTokenPrice : outputTokenPrice)
-
-      const priceImpact = getPriceImpact(
-        isMinting ? quoteAmountUsd : outputTokenAmountUsd,
-        isMinting ? outputTokenAmountUsd : quoteAmountUsd,
-      )
-
-      const outputTokenAmountUsdAfterFees = outputTokenAmountUsd - gas.costsUsd
-
       const fullCostsInUsd = getFullCostsInUsd(
         inputAmount,
         gas.limit * gas.price,
         inputToken.decimals,
-        inputTokenPrice,
+        quoteFM.inputTokenPrice,
         ethPrice,
       )
 
-      // includes swap fee (which we can't distinguish for now)
-      const mintRedeemFees = isMinting ? fees.mint : fees.redeem
-      const mintRedeemFeesUsd = inputTokenAmountUsd * mintRedeemFees
-      const priceImpactUsd =
-        (isMinting ? quoteAmountUsd : inputTokenAmountUsd) -
-        (isMinting ? outputTokenAmountUsd : quoteAmountUsd) -
-        mintRedeemFeesUsd
-      const priceImpactPercent = (priceImpactUsd / inputTokenAmountUsd) * 100
+      const isHighPriceImpact = quoteFM.priceImpactPercent > 2
 
-      const isHighPriceImpact = priceImpactPercent > 2
+      const outputTokenAmountUsdAfterFees =
+        quoteFM.outputAmountUsd - gas.costsUsd
 
       return {
+        ...quoteFM,
```

Comment:
The quote API response now comes with more data that was previously generated on the client (see below)